java8/src/main/java/net/nycjava/java8/api/J8OptionalApi.java

```diff
@@ -1,41 +1,23 @@
 package net.nycjava.java8.api;
 
-import java.util.AbstractMap.SimpleEntry;
-import java.util.List;
-import java.util.Map;
 import java.util.Optional;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.unmodifiableMap;
-import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.joining;
 
 public class J8OptionalApi {
 
 
     public static void main(String[] args) {
+        System.out.println(concatName("Douglas", null, "Crockford"));
+        System.out.println(concatName("Guido", "Van", null));
```

Comment:
I can't imagine someone who has a first and middle name without a last name (AKA family name).
perhaps a better example would be someone with a single name, like "Madonna", or "Prince" or some other pop icon.

Reply:
yep